qutebrowser/completion/models/sortfilter.py

```diff
@@ -140,21 +133,33 @@ def filterAcceptsRow(self, row, parent):
         if parent == QModelIndex() or not self.pattern:
             return True
 
-        try:
-            return self.srcmodel.custom_filter(self.pattern, row, parent)
-        except NotImplementedError:
-            for col in self.srcmodel.columns_to_filter:
-                idx = self.srcmodel.index(row, col, parent)
-                if not idx.isValid():
-                    # No entries in parent model
-                    continue
-                data = self.srcmodel.data(idx)
-                if not data:
-                    continue
-                elif self.pattern_re.search(data):
-                    return True
+        # first check the cache
```

Comment:
I don't understand the details, but the `layoutChanged` signal emited in `invalidate` is handled in [`_q_sourceLayoutChanged`](https://github.com/qt/qtbase/blob/5.7.0/src/corelib/itemmodels/qsortfilterproxymodel.cpp#L1356-L1389), which calls [`update_persistent_indexes`](https://github.com/qt/qtbase/blob/5.7.0/src/corelib/itemmodels/qsortfilterproxymodel.cpp#L1036-L1053) -> [`create_mapping`](https://github.com/qt/qtbase/blob/5.7.0/src/corelib/itemmodels/qsortfilterproxymodel.cpp#L328-L372) -> `filterAcceptsRow`.